### ISSUE TITLE
Release v0.3

### ISSUE THREAD

**Issue by adam2392** [mne-tools/mne-icalabel#90]
We should release v0.3 once #88 and #73 is merged.

**Comment by mscheltienne** [comment]
Also, we will need to update the conda-forge recipe with the 'new' dependencies.
We could do something like MNE with:
- `conda install -c conda-forge mne-icalabel` equivalent to `pip install mne-icalabel[ica,gui]`
- `conda install -c condaa-forge mne-icalabel-dev` equivalent to `pip install mne-icalabel[all]`

**Comment by adam2392** [comment]
Do you remember how to update the conda recipe? Is it this file? Are there writeup instructions anywhere?

https://github.com/conda-forge/mne-icalabel-feedstock/blob/main/recipe/meta.yaml

**Comment by mscheltienne** [comment]
Yes, it's the recipe. I know how to do it, and I'll fix it after the Pypi release. Even if the conda-bot picks up the release, it won't automerge the PR as it will detect changes in the dependencies. 

Also, now that I'm thinking about it, the `mne-icalabel-dev` key won't work. Just as MNE does not have `-dev` key, most of the development dependencies are not available on conda.

**Comment by adam2392** [comment]
I think perhaps we leave conda as just the base install for full functionality perhaps?

E.g. base and GUI? 

Cuz everything else is technically dev related and can be added with pip?

**Comment by mscheltienne** [comment]
I think we should have this:

- `conda install -c conda-forge mne-icalabel` equivalent to `pip install mne-icalabel[ica,gui]`

------------

2 scenarios:
- Someone installs MNE and then MNE-ICALabel

```
conda install -c conda-forge mne
conda install -c conda-forge mne-icalabel
```

Both GUI and ICA dependencies are already resolved by the installation of MNE https://github.com/conda-forge/mne-feedstock/blob/ced43efa6f73d7a3438301c6ac30e48987c46e0a/recipe/meta.yaml#L53-L90

- Someone install directly MNE-ICALabel without installing MNE first

```
conda install -c conda-forge mne-icalabel
```

This one will only install `mne-base` which does not include the GUI and ICA dependencies https://github.com/conda-forge/mne-icalabel-feedstock/blob/48843682a6dedc1843e5070b447edf9061eb844c/recipe/meta.yaml#L26
So, as for MNE which installs all the extra-dependencies when you do `conda install -c conda-forge mne`, let's also add the dep. for GUI and ICA here.

-----------

For `mne-icalabel-base`, I was thinking about the MNE-installers, but even in their recipes it does not make sense to exclude the extra-dep we (might) need.

**Comment by mscheltienne** [comment]
I think we don't need to wait on #73, I'm still not convinced it is ready to be merged. But let's wait on #93, at least because of the format change from array to list for `ica.labels_`.

**Comment by adam2392** [comment]
I'm not in a particular rush, so I'm okay w/ waiting for #73 and #93 since the next steps of the work I think are:

1. annotating the dataset once or twice when Aaron returns from his summer camp
2. begin testing out models for end-to-end training
3. benchmark models w/ out of the box stuff like RF

These will probably require some of the feature stuff anyways.

**Comment by adam2392** [comment]
@hoechenberger just to clarify, for me to update the conda recipe I have to make a PR to the feedstock meta.yml file?

**Comment by mscheltienne** [comment]
I can do it if you want, or we can wait for the bot to pick it up, create the PR, and ask us to fix it because some of the dependencies changed.

**Comment by adam2392** [comment]
Released v0.3 on pypi

**Comment by adam2392** [comment]
Will close now when conda picks up v0.3
